src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -81,8 +91,71 @@ public MasterIndexFieldByRecordClassSchemaResponse getMasterIndexFieldByRecordCl
+                                                                   .append(ADD_ASYNC_SEARCH_RELATED_TASK_NAME)
+                                                                   .toString(),
+                                                               executionDetail, userAccount));
+
+        AddAsyncSearchRequestGenerator requestGenerator;
+        try {
+            requestGenerator = createAddAsyncSearchRequestGenerator(searchName, executionDetail, armAutomatedTaskEntity, now);
+        } catch (NullPointerException e) {
+            throw handleFailureAndCreateException(exceptionMessageBuilder.append("Could not construct API request: ")
+                                                      .append(e)
+                                                      .toString(),
+                                                  executionDetail, userAccount);
+        }
+
+        ArmAsyncSearchResponse response;
+        try {
+            response = armRpoClient.addAsyncSearch(bearerToken, requestGenerator.getJsonRequest());
+        } catch (FeignException e) {
+            throw handleFailureAndCreateException(exceptionMessageBuilder.append("API call failed: ")
+                                                      .append(e)
+                                                      .toString(),
+                                                  executionDetail, userAccount);
+        }
+
+        String searchId = response.getSearchId();
+        if (searchId == null) {
+            throw handleFailureAndCreateException(exceptionMessageBuilder.append("The obtained search id was empty")
+                                                      .toString(),
+                                                  executionDetail, userAccount);
+        }
+
+        executionDetail.setSearchId(searchId);
+        armRpoService.updateArmRpoStatus(executionDetail, ArmRpoHelper.completedRpoStatus(), userAccount);
+    }
+
+    private AddAsyncSearchRequestGenerator createAddAsyncSearchRequestGenerator(String searchName,
+                                                                                       ArmRpoExecutionDetailEntity executionDetail,
+                                                                                       ArmAutomatedTaskEntity armAutomatedTaskEntity,
+                                                                                       OffsetDateTime now) throws NullPointerException {
+        return AddAsyncSearchRequestGenerator.builder()
+            .name(searchName)
+            .searchName(searchName)
+            .matterId(executionDetail.getMatterId())
+            .entitlementId(executionDetail.getEntitlementId())
+            .indexId(executionDetail.getIndexId())
+            .sortingField(executionDetail.getSortingField())
+            .startTime(now.minusHours(armAutomatedTaskEntity.getRpoCsvStartHour()))
+            .endTime(now.plusHours(armAutomatedTaskEntity.getRpoCsvEndHour()))
```

Comment:
@hemantasharma1129 for the `endTime` field (which maps to the second item in the `value` array in the request), the detail on [DMP-3616](https://tools.hmcts.net/jira/browse/DMP-3616) has:
> "2024-08-13T23:00:00.000Z" : this is the value current datetime minus rpo_csv_end_hour  from arm_automated_task table.

Is that statemet on the ticket correct? I've assumed this should be "current datetime PLUS rpo_csv_end_hour" as that made more sense to me, but I could be wrong.

Can you confirm which is correct please?

Reply:
@chrisbellingham-hmcts yeah thats correct. Start hour should be currentdatetime minus Db end hour and End hour should be currentdatetime minus Db start hour